### PR TITLE
[MRG] anonymize sex and hand when keep_his is false

### LINKED ISSUE #8130: anonymize also sex and hand
anonymize with keep_his False keeps what's in: `raw.info['subject_info']`
which contains hand and sex for example.

Should we remove these info with keep_his=False or should we introduce another parameter?

### PR DESCRIPTION
**Pull request by Falach**
#### Reference issue
Fixes #8130

#### What does this implement/fix?
when keep_his = false it's also anonymize hand and sex

#### Additional information
I updated the relevant tests as well

### CODE REVIEW

**Review by drammock — APPROVED** [review]
LGTM.  I think the Circle failure is one we already fixed on main (@larsoner is that right?)

In which case this should be good to go.

**Review by sappelhoff — APPROVED** [review]
lgtm and I think it's a good idea to drop all HIS info if "keep_his" is False.

I didn't look into the circle issue, assuming that it's fixed in main.

**Comment by drammock** [comment]
I've pushed a commit that merges the current `upstream/main` branch into this branch, to see if it fixes the CircleCI error.  @Falach if you need to do any more updates to this branch, you should first do
```
git fetch origin
git merge origin/anonymize_his
```

**Review by agramfort — COMMENTED** [review]
can you just add a what's new entry to document this change? thx @Falach

**Comment by sappelhoff** [comment]
@Falach you have a circleci issue:

> changes/latest.inc:120: WARNING: Unknown target name: "rotem falach".

The problem is that sphinx doesn't know about your name yet. Could you please add it to this file (roughly alphabetically)?

https://github.com/mne-tools/mne-python/blob/5f0beb93f325ecd53e88fca500f1d08fbead863e/doc/changes/names.inc#L139

And if this is your first entry in the changelog, please also add a mapping of your name like here:

https://github.com/mne-tools/mne-python/blob/5f0beb93f325ecd53e88fca500f1d08fbead863e/doc/changes/latest.inc#L17-L40

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -113,6 +113,8 @@ Enhancements
 
 - `mne.viz.plot_drop_log` and :meth:`mne.Epochs.plot_drop_log` now omit displaying the subject name in the title if ``subject=None`` is passed (:gh:`9015` by `Richard Höchenberger`_)
 
+- `mne.io.meas_info.anonymize_info` now anonymize also sex and hand fields when keep_his is False (:gh:`8130` by `Rotem Falach`_)`
```
Comment (sappelhoff) [inline doc/changes/latest.inc:116]: once you implemented my requests from https://github.com/mne-tools/mne-python/pull/9103#issuecomment-801133563 you will need to adjust this line:

```suggestion
- `mne.io.meas_info.anonymize_info` now anonymize also sex and hand fields when keep_his is False (:gh:`8130` by |Rotem Falach|_)`
```

**Comment by Falach** [comment]
> I've pushed a commit that merges the current `upstream/main` branch into this branch, to see if it fixes the CircleCI error. @Falach if you need to do any more updates to this branch, you should first do
> 
> ```
> git fetch origin
> git merge origin/anonymize_his
> ```
> 
> before making more local changes

I swear I did it and it failed again! :(

File: doc/changes/latest.inc [hunk 2]
```diff
@@ -113,6 +113,8 @@ Enhancements
 
 - `mne.viz.plot_drop_log` and :meth:`mne.Epochs.plot_drop_log` now omit displaying the subject name in the title if ``subject=None`` is passed (:gh:`9015` by `Richard Höchenberger`_)
 
+- `mne.io.meas_info.anonymize_info` now anonymize also sex and hand fields when keep_his is False (:gh:`9103` by |Rotem Falach|_)`
```
Comment (agramfort) [inline doc/changes/latest.inc:116]:
```suggestion
- :func:`mne.io.meas_info.anonymize_info` now anonymizes also sex and hand fields when ``keep_his`` is ``False`` (:gh:`9103` by |Rotem Falach|_)`
```

**Comment by sappelhoff** [comment]
@Falach it fails because you have not yet implemented the changes I describe here: https://github.com/mne-tools/mne-python/pull/9103#issuecomment-801133563

**Comment by drammock** [comment]
I think this will still fail, because our changelog is actually quite complicated (sorry).  You will need to add yourself to `doc/changes/names.inc`, and also at the top of `doc/changes/latest.inc` add a special replacement line for new contributors.  @Falach if you're having trouble getting the local branch in sync with this PR, let me know and I can push the necessary changes.

**Comment by sappelhoff** [comment]
@Falach you can look at the commits ebf1dc4 and 0429750 that I just made on your branch. this is what I tried to express with my comment https://github.com/mne-tools/mne-python/pull/9103#issuecomment-801133563 - sorry if that didn't come across.

Dan is right, our changelog is slightly complicated :-)

**Review by agramfort — APPROVED** [review]
can you just see why CIs are red?

thx @Falach !

File: doc/changes/latest.inc [hunk 3]
```diff
@@ -113,6 +115,8 @@ Enhancements
 
 - `mne.viz.plot_drop_log` and :meth:`mne.Epochs.plot_drop_log` now omit displaying the subject name in the title if ``subject=None`` is passed (:gh:`9015` by `Richard Höchenberger`_)
 
+- :func:`mne.io.meas_info.anonymize_info` now anonymizes also sex and hand fields when ``keep_his`` is ``False`` (:gh:`9103` by |Rotem Falach|_)`
```
Comment (sappelhoff) [inline doc/changes/latest.inc:118]: just a minor issue with the module import here:

```suggestion
- :func:`mne.io.anonymize_info` now anonymizes also sex and hand fields when ``keep_his`` is ``False`` (:gh:`9103` by |Rotem Falach|_)`
```

**Review by sappelhoff — APPROVED** [review]
LGTM, I found a tiny issue in `latest.inc` that I am going to fix here with a suggestion.

I think we can merge afterwards.

**Comment by agramfort** [comment]
thx @Falach

**Pull request merged**